rfcs/0180-broken-package-removal.md

```diff
@@ -0,0 +1,85 @@
+---
+feature: broken-package-removal
+start-date: 2024-07-14
+author: Jörg "Mic92" Thalheim
+co-authors:
+shepherd-team:
+shepherd-leader:
+related-issues:
+---
+
+# Summary
+
+[summary]: #summary
+
+This RFCs defines under what conditions we remove broken and unmaintained leaf packages.
+The RFC does not have the ambition to define all reasons to remove a package but
+instead focuses on simple rules that can be automated.
+
+# Motivation
+
+[motivation]: #motivation
+
+Broken and unmaintained packages still cost us valuable time.
+Broken packages still have to be evaluated to some extend
+and unmaintained packages might still have to rebuild when one of their dependencies changes.
+Removing unmaintained/unused code hopefully increases the overall package quality and helps us
+to save resources.
+Because it is easy to add packages, it should be also easy to remove them, ideally automated.
+
+# Detailed design
+
+[design]: #detailed-design
+
+## Broken Packages
```

Comment:
I suggest to create some archive for them both broken and unmaintained.

A bot can save these deleted expressions to another repo, that would be similar to an [overlay](https://wiki.nixos.org/wiki/Overlays).

Reply:
(well, I can do this on [Museoa](https://github.com/museoa/bqn-mode) just because...)

Reply:
I don't think a bot can do this as it often requires more work to keep packages in a way that also evaluate. However what we can do is link to the pull request that removed the package. This makes it easy to find the code and the nixpkgs revision that still contained the package.

Reply:
However if you want to do this. Feel free. I don't think this RFC will conflict with this. Depending on how you want to implement this we can also establish some notification mechanism.